src/main/ruby/core/array.rb

```diff
@@ -475,7 +475,8 @@ def fetch(idx, default=undefined)
       end
     end
 
-    if left >= Fixnum::MAX || right > Fixnum::MAX
+    unless Truffle.invoke_primitive(:fixnum_fits_into_long, left) &&
```

Comment:
Was it significant that left was a `>=` check while right was a `>` check?

Reply:
Doesn't look like so, I think it was a bug.

Reply:
Actually, it looks like `left` is always checked by `coerce_to_collection_length`, which already checks that it is a Fixnum.
Same for `right`, so I should be able to remove that check completely.

Reply:
I misread, those values are still modified after `coerce_to_collection_length`.
So I think it was indeed a bug to use `>=`.